### PR TITLE
Capstone

### PR DESCRIPTION
**Pull request by weaversa**

### CODE REVIEW

**Comment by weaversa** [comment]
This is a kind of capstone lab that includes parts of many of the labs here. I'd love comments!!!

**Comment by weaversa** [comment]
We envision that when the lab is completed, the message decrypted by the student would be something special that celebrates their completion of the Cryptol course. This has yet to be designed.

**Comment by weaversa** [comment]
> @weaversa Was the `# Computational Requirements` section of `LoremIpsum.md` supposed to not be word-wrapped?

I haven’t viewed the lab in Markdown yet. Feel free to make a commit for any silly things you see.

File: labs/LorumIpsum/LorumIpsum.md [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+Cyclic Redundancy Checks (CRC) are used to add integrity to the Lorum
+Ipsum key at two stages of generation, namely, after key wrapping and
+after tagging.
+
+
+## CRC-32
+
+The specific variant of CRC used is the CRC-32c defined in
+[RFC 3309](https://tools.ietf.org/pdf/rfc3309.pdf). For reference, the
+polynomial is given as,
```
Comment (AssemblyZig) [inline labs/LorumIpsum/LorumIpsum.md:209]: ',' --> ':'

File: labs/LorumIpsum/LorumIpsum.md [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+
+
+# Key Wrap Specification
+
+The Lorum Ipsum key is wrapped using the KW-AE 128 Algorithm described
+in [NIST Special Publication
+800-38F](https://csrc.nist.gov/publications/detail/sp/800-38f/final)
+"Recommendation for Block Cipher Modes of Operation: Methods for Key
+Wrapping". This algorithm takes a 128-bit KEK and 128-bit key and
+produces 192-bits of wrapped key.
```
Comment (AssemblyZig) [inline labs/LorumIpsum/LorumIpsum.md:252]: "192-bits" --> "192 bits"

File: labs/LorumIpsum/LorumIpsum.md [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+given 3-ASCII-digit key issue number will be prepended to the 192-bit
+wrapped key after the first CRC-32c has been applied.
+
+
+### Valid Issue Numbers
+
+The KLI20 will accept any 24-bit issue number. However, this key
+specification issue only uses numbers that correspond to the ASCII
+strings between and including "000" to "999". Specifically, this key
+specification follows a special schedule that is comprised entirely of
```
Comment (AssemblyZig) [inline labs/LorumIpsum/LorumIpsum.md:340]: "comprised" --> "composed"  https://www.businesswritingblog.com/business_writing/2015/01/comprised-of-vs-composed-of-a-test.html

File: labs/LorumIpsum/LorumIpsumAnswers.md [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+Cyclic Redundancy Checks (CRC) are used to add integrity to the Lorum
+Ipsum key at two stages of generation, namely, after key wrapping and
+after tagging.
+
+
+## CRC-32
+
+The specific variant of CRC used is the CRC-32c defined in
+[RFC 3309](https://tools.ietf.org/pdf/rfc3309.pdf). For reference, the
+polynomial is given as,
```
Comment (AssemblyZig) [inline labs/LorumIpsum/LorumIpsumAnswers.md:216]: , --> :

File: labs/LorumIpsum/LorumIpsumAnswers.md [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+> Like most key specs, this is _mostly_ clear after thinking hard
+  about what it says and looking at Figure 1.
+
+```
+tag : [3][8] -> [224] -> [352]
+tag Issue x = join "LoremIpsumKey" # (join Issue) # x
+```
+
+> Now that we have all the helper function (wrapping, tagging, and
+  CRC), we can look at Figure 1 and tie the whole thing together.
```
Comment (AssemblyZig) [inline labs/LorumIpsum/LorumIpsumAnswers.md:357]: "helper function" --> "helper functions"

File: labs/LorumIpsum/LorumIpsumAnswers.md [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+```
+secretMessageIssue0 = 0x0d84af8336884da53714f6eafa2bf80f38e5028d208ce39c8d78a1f768738413aa577598816241adf3077e2587ffbd7456e8583a2155e6411dc64a15e414bafc2556184488a353c54d5f274c2b54eef4ef7ebd8d9a7f13137e6bacf1b7ff605dd55443d7a980e7ba298919fd432be0082699ece1d8c1f8e0849bcc1beb07a2c005c622ae1e5fe79e43ae31f0a1c13b9baa045ceeb9a43ee47a7d09702187c8f8ab51309c308f1b7d22ac2bc2da49487c88ebd527127a709da2418c35
+
+secretMessageIssue1 = 0x7f0f165e95c728ab1d1c07aa3c12cc10d5a975394c37eb5870be8b5495334ff472c5192b9f97faea063540b5f11d51fdbbdb4117ea9612cc0c6b42c3b70dab7615cd
+
+secretMessageIssue2 = 0x4ba6d9eb8489faed2223bd5e3bbf5bf313708c38b369b0fea673c76843cf5312252984b024c2f4263ef318dfbade320920558cb7e475c54811f955a612ab9e18128e1adb056e5775
+```
+
+> We leave this exercise to the student. Don't forget to set `:s
+  ascii=on`.
```
Comment (AssemblyZig) [inline labs/LorumIpsum/LorumIpsumAnswers.md:453]: Perhaps more succinctly "Don't forget to `:set ascii=on`....

File: labs/LorumIpsum/LorumIpsum.md [hunk 7]
```diff
@@ -0,0 +1,404 @@
+# LorumIpsum Key
+
+During this lab you will use the results of previous labs to
+Cryptolize a *mock* key generation specification we're calling the
+LorumIpsum key. This document specifies the algorithm, but, like most
+specification documents it also contains reams of text unrelated to
+the algorithm. We've filled in these unhelpful portions with [lorum
+ipsum](http://www.lipsum.com/). You may happily skip over them to get
+to the important bits.
```
Comment (AssemblyZig) [inline labs/LorumIpsum/LorumIpsum.md:9]: Do you wanna also italicize the *lorem ipsum* sections?  Just a thought....

File: labs/LorumIpsum/LorumIpsumAnswers.md [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+
+
+# Key Wrap Specification
+
+The Lorum Ipsum key is wrapped using the KW-AE 128 Algorithm described
+in [NIST Special Publication
+800-38F](https://csrc.nist.gov/publications/detail/sp/800-38f/final)
+"Recommendation for Block Cipher Modes of Operation: Methods for Key
+Wrapping". This algorithm takes a 128-bit KEK and 128-bit key and
+produces 192-bits of wrapped key.
```
Comment (AssemblyZig) [inline labs/LorumIpsum/LorumIpsumAnswers.md:260]: "192-bits" --> "192 bits"

File: labs/LorumIpsum/LorumIpsumAnswers.md [hunk 9]
```diff
@@ -0,0 +1,453 @@
...
+    TaggedKey  = tag Issue CRCWK
+    CRCTK      = TaggedKey # (CRC32_C (split TaggedKey))
+```
+
+### Valid Issue Numbers
+
+The KLI20 will accept any 24-bit issue number. However, this key
+specification issue only uses numbers that correspond to the ASCII
+strings between and including "000" to "999". Specifically, this key
+specification follows a special schedule that is comprised entirely of
```
Comment (AssemblyZig) [inline labs/LorumIpsum/LorumIpsumAnswers.md:374]: "comprised" --> "composed"

**Pull request merged**